lib/AST/Decl.cpp

```diff
@@ -719,7 +720,14 @@ Optional<CustomAttrNominalPair> Decl::getGlobalActorAttr() const {
 }
 
 bool Decl::predatesConcurrency() const {
-  return getAttrs().hasAttribute<PredatesConcurrencyAttr>();
+  if (getAttrs().hasAttribute<PredatesConcurrencyAttr>())
+    return true;
+
+  // Imported C declarations always predate concurrency.
+  if (isa<ClangModuleUnit>(getDeclContext()->getModuleScopeContext()))
+    return true;
```

Comment:
Out of curiosity, what will this match that `hasClangNode()` wouldn't or vice versa?

Reply:
IIRC, there's some cases involving synthesized extensions for import-as-member that differ.